streaming-twitter/src/main/scala/org/apache/spark/streaming/twitter/TwitterInputDStream.scala

```diff
@@ -85,10 +85,8 @@ class TwitterReceiver(
         }
       })
 
-      val query = new FilterQuery
-      if (filters.size > 0) {
-        query.track(filters.mkString(","))
-        newTwitterStream.filter(query)
+      if (query.isDefined) {
+        newTwitterStream.filter(query.get)
       } else {
```

Comment:
As I mentioned in the PR description, the limitation of hiding the FilterQuery from the user is that we are only able to filter the Twitter stream via disjunctive keyword queries:

```scala
// this will give us any Tweet that contains "foo", "bar" or "baz"
val tweets = TwitterUtils.createStream(ssc, Seq("foo", "bar", "baz"));
```

However, the Twitter stream API also supports many other types of filtering, including:
- Receive Tweets that are tagged at a particular location (ref: [locations](https://dev.twitter.com/streaming/overview/request-parameters#locations))
- Receive Tweets created by specific users (ref: [follow](https://dev.twitter.com/streaming/overview/request-parameters#follow))
- Receive Tweets that match a conjunction of keywords (ref: [track with spaces](https://dev.twitter.com/streaming/overview/request-parameters#track))

Refer to Twitter's [official documentation](https://dev.twitter.com/streaming/overview/request-parameters) for a full list of all supported filters.

By exposing the FilterQuery, we enable users to make use of all of these powerful filters and any future filters that Twitter may introduce.